### PR TITLE
fix: allow to really overwrite vnet_subnet_id in extra specs

### PR DESCRIPTION
**Pull request by maigl**
The two fields 'vnet_subnet_id' and 'disable_network_isolation' could be defined in extra specs but would not be used to overwrite the defaults define in the config.

I also added a unit test for this.

This unit test is by far not complete in terms of testing all config vs extraspec behavior.

I also added a `-version` flag that can be used for binary stamping.

<sub>Michael Kuhnt <michael.kuhnt@mercedes-benz.com> Mercedes-Benz Tech Innovation GmbH ([ProviderInformation](https://github.com/mercedes-benz/daimler-foss/blob/master/PROVIDER_INFORMATION.md))
</sub>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -32,7 +33,19 @@ var signals = []os.Signal{
 	syscall.SIGTERM,
 }
 
+// Version is the version of the provider.
```
Comment (gabriel-samfira) [inline main.go:36]: I was thinking of extending the garm external provider interface and adding new commands like `GARM_COMMAND=GetVersionInformation`.

Command that would return a json which contains something like:

```json
{
  "provider_version": "v0.1.0",
  "provider_commit_id": "SHA_GOES_HERE",
  "build_time": "not sure it's worth it but might be a good indicator I guess",
  "provider_interface_version": "v0.2.0",
  "supported_interface_versions": ["v0.1.0", "v0.2.0"]
}
```

Not sure yet what else would be useful. What do you think?
And some more fields I can't think of right now.

Comment (maigl) [reply main.go:36]: .. sounds fine .. and all data is binary stamped into the provider?

shall I add this ? or shall I just remove me lame -version flag?

Comment (gabriel-samfira) [reply main.go:36]: The version srting and the SHA, yes. The rest should be added as actual code. The reason I mention this is that I'd rather omit adding the version flag parsing for now, and add it later as part of a larger change to the garm ecosystem.

Comment (maigl) [reply main.go]: okay .. I dropped the commit

**Review by gabriel-samfira — APPROVED** [review]
LGTM!

**Pull request merged**